### PR TITLE
Refactored payment info card section in Order Detail

### PR DESCRIPTION
**Pull request by anitaa1990**
This PR takes care of the fourth step of the Order Detail refactoring #2844 .

#### Changes
- Adds the Order Payment Info card (`OrderDetailPaymentInfoView`) to the Order detail screen.
- Adds 2 new variables to the `Order` model to find out if an order is paid or is awaiting payment.

#### Notes
- I have split this massive change into multiple small PRs in order to make reviews easier. 
- This PR is to be merged into a feature branch.
- This PR just adds the Order detail Payment info card to the Order detail screen. The remaining cards will be added in subsequent PRs along with unit tests and click actions for all the buttons in the Order detail.
- **This PR is in draft till this [Step 3 PR](https://github.com/woocommerce/woocommerce-android/pull/2860) can be merged.**

#### Testing
- Click on an order from the Orders tab.
- Verify that the Payment card is displayed correctly for an order. **Note that the `Refunds` section of the `Payment` card will take place in a subsequent PR since it requires information from the refunds API.**

#### Screenshots
. 

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/views/OrderDetailPaymentInfoView.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        if (order.isOrderPaid) {
+            paymentInfo_paymentMsg.hide()
+            paymentInfo_paidSection.hide()
+        } else {
+            paymentInfo_paymentMsg.show()
+
+            if (order.isAwaitingPayment) {
+                paymentInfo_paid.text = formatCurrencyForDisplay(BigDecimal.ZERO) // Waiting for payment
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/views/OrderDetailPaymentInfoView.kt]: NP: The comment "Waiting for payment" seems unnecessary here.

**Comment by nbradbury** [comment]
@anitaa1990 Changes look good, but it looks like the address line fix from a previous PR hasn't been pulled into this PR.

Screenshot_1600688284

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/views/OrderDetailPaymentInfoView.kt (hunk 1 above)
Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/views/OrderDetailPaymentInfoView.kt]: 👍 Fixed in 079eed9

**Comment by anitaa1990** [comment]
Thanks @nbradbury! I pulled in the changes from the feature branch into this PR. Ready for another round 👍

**Review by nbradbury — APPROVED** [review]
:shipit:

**Pull request merged**